### PR TITLE
Github Actions

### LINKED ISSUE #23: NuGet publishing
Publish our OSS packages to NuGet. Individual issues for the same: https://github.com/Lombiq/Helpful-Extensions/issues/34, https://github.com/Lombiq/Orchard-Azure-Application-Insights/issues/15, https://github.com/Lombiq/UI-Testing-Toolbox/issues/98

First steps:

- Set up the NuGet publishing with GitHub Actions of https://github.com/Lombiq/Helpful-Libraries and https://github.com/Lombiq/Helpful-Extensions. These will then at a later point serve as models for the rest.
- These two projects are special, and thus cover all important use cases, because Helpful Extensions depends on Helpful Libraries, and Helpful Libraries contains three csprojs that all need to be published (either as separate packages or one; perhaps the former because other projects may not depend on all of them).
- We're developing such OSS projects as part of this OSOCE solution. In other projects of ours we use them as submodules too, so consuming and improving them is easier. We'll keep this, thus we need some way for this to not clash with NuGet references.

Some criteria to keep in mind:

- We can handle the whole thing from a build in OSOCE too somehow, it needn't be for each project, if it makes things easier.
- Currently, we have about 30 projects that we'd like to publish to NuGet. This already necessitates the ability to manage NuGet publishing somehow centrally in a DRY manner (apart from credentials that can be stored in GitHub secrets) but we'll likely have many more. So, a solution where we need to maintain a pipeline in each of these repositories would be an issue

### PR DESCRIPTION
**Pull request by deanmarcussen**
Fixes https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/issues/23

Merge 3rd.

This depends on the `NuGet.config` to get HL from Cloudmsith.

(the version specified is currently available there, as I pushed it from my fork)

### CODE REVIEW

File: Lombiq.HelpfulExtensions.csproj [hunk 1]
```diff
@@ -35,4 +35,8 @@
       Condition="Exists('..\..\Libraries\Lombiq.HelpfulLibraries\Lombiq.HelpfulLibraries.csproj')" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(NuGetBuild)' == 'True'">
+    <PackageReference Include="Lombiq.HelpfulLibraries.Targets" Version="0.0.3" />
```
Comment (deanmarcussen) [inline Lombiq.HelpfulExtensions.csproj:36]: this will need updating after merging and building HL

File: NuGet.config [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Piedone) [inline NuGet.config:1]: Do we still need this file?

File: Readme.md [hunk 3]
```diff
@@ -69,3 +69,10 @@ It displays an [accordion powered by Bootstrap](https://getbootstrap.com/docs/4.
 Bug reports, feature requests, comments, questions, code contributions, and love letters are warmly welcome, please do so via GitHub issues and pull requests. Please adhere to our [open-source guidelines](https://lombiq.com/open-source-guidelines) while doing so.
 
 This project is developed by [Lombiq Technologies](https://lombiq.com/). Commercial-grade support is available through Lombiq.
+
+[![Hosted By: Cloudsmith](https://img.shields.io/badge/OSS%20hosting%20by-cloudsmith-blue?logo=cloudsmith&style=for-the-badge)](https://cloudsmith.com)
+
+Package repository hosting is graciously provided by  [Cloudsmith](https://cloudsmith.com).
+Cloudsmith is the only fully hosted, cloud-native, universal package management solution, that
+enables your organization to create, store and share packages in any format, to any place, with total
+confidence.
```
Comment (Piedone) [inline Readme.md:78]: Note to self: Remove after update to NuGet.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: NuGet.config (hunk 2 above)
Comment (deanmarcussen) [reply NuGet.config:1]: as per your comment below:
> Note to self: Remove after update to NuGet.

**Review by Piedone — APPROVED** [review]

**Pull request merged**